sdk/forms/src/test/java/com/klaviyo/forms/KlaviyoWebViewDelegateTest.kt

```diff
@@ -0,0 +1,380 @@
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        // notably no message received
+        staticClock.execute(10_000)
+
+        verify { spyLog.debug("IAF WebView Aborted: Timeout waiting for Klaviyo.js") }
+        verifyClose()
+    }
+
+    @Test
+    fun `postMessage handles close correctly`() {
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage("""{"type":"formDisappeared"}""")
+
+        verifyClose()
+    }
+
+    @Test
+    fun `abort messages properly closes`() {
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage("""{"type":"abort"}""")
+
+        verify { spyLog.info("IAF aborted, reason: Unknown") }
+        verifyClose()
+    }
+
+    @Test
+    fun `create profile event handled properly`() {
+        val eventMessage = """
+           {
+              "type": "trackProfileEvent",
+              "data": {
+                "metric": "Form completed by profile",
+                "properties": {}
+              }
+           } 
+        """.trimIndent()
+        val expectedMetric = EventMetric.CUSTOM("Form completed by profile")
+
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage(eventMessage)
+
+        // required steps to the profile event
+        verify { mockState.getAsProfile() }
+        val slot = slot<Event>()
+        verify { mockApiClient.enqueueEvent(capture(slot), any()) }
+        assertEquals(slot.captured.metric, expectedMetric)
+    }
+
+    @Test
+    fun `verify webview closes on an orientation change`() {
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+
+        slotOnActivityEvent.captured(ActivityEvent.ConfigurationChanged(Configuration()))
+        // if we emit the same config change we still should only close once
+        slotOnActivityEvent.captured(ActivityEvent.ConfigurationChanged(Configuration()))
+
+        verify(exactly = 1) { spyLog.debug("New screen orientation, closing form") }
+        verifyClose()
+    }
+
+    @Test
+    fun `verify close fails on a null webview`() {
+        val delegate = KlaviyoWebViewDelegate()
+        // notably do not init webview
+        delegate.postMessage("""{"type":"formDisappeared"}""")
+        verify { spyLog.warning("Unable to close IAF - null WebView reference") }
+        verifyClose(true)
+    }
+
+    @Test
+    fun `verify close fails on a null decorview`() {
+        every { mockActivity.window?.decorView } returns null
+
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage("""{"type":"formDisappeared"}""")
+        verify { spyLog.warning("Unable to close IAF - null activity reference") }
+        verifyClose(true)
+    }
+
+    @Test
+    fun `postMessage handles AggregateEventTracked`() {
```

Comment:
nit: some of these might be a bit duplicative of tests in `BridgeMessageTest`

Reply:
Meh i guess that doesn't really matter though. Except maybe nice for DRYness to pull some shared strings out into a fixtures file.